src/libcollections/str.rs

```diff
@@ -849,10 +854,18 @@ impl str {
     #[unstable(feature = "str_utf16",
                reason = "this functionality may only be provided by libunicode",
                issue = "27714")]
+    #[rustc_deprecated(since = "1.8.0", reason = "renamed to encode_utf16")]
```

Comment:
Isn't that just for deprecating stable APIs? For deprecating unstable APIs everyone should be able to immediately switch, right?